src/webgpu/api/validation/resource_usages/textureUsageInRender.spec.ts

```diff
@@ -49,64 +61,173 @@ class TextureUsageTracking extends ValidationTest {
-    {
-      writeBaseLevel: READ_BASE_LEVEL + 1,
-      writeBaseLayer: READ_BASE_LAYER,
-      _success: true,
-    },
-
-    // read and write usages are binding to different array layers of the same texture.
-    {
-      writeBaseLevel: READ_BASE_LEVEL,
-      writeBaseLayer: READ_BASE_LAYER + 1,
-      _success: true,
-    },
-  ])
-  .fn(async t => {
-    const { writeBaseLevel, writeBaseLayer, _success } = t.params;
+const BASE_LEVEL = 3;
+const BASE_LAYER = 0;
+const TOTAL_LEVELS = 6;
+const TOTAL_LAYERS = 2;
 
-    const texture = t.createTexture({ arrayLayerCount: 2, mipLevelCount: 6 });
+g.test('subresources_and_binding_types_combination_for_color')
+  .params(
+    params()
+      .combine([
+        // Two texture usages are binding to the same texture subresource.
+        {
+          baseLevel: BASE_LEVEL,
+          baseLayer: BASE_LAYER,
+          levelCount: 1,
+          layerCount: 1,
+          _resourceSuccess: false,
+        },
 
-    const sampleView = texture.createView({
-      baseMipLevel: READ_BASE_LEVEL,
-      mipLevelCount: 1,
-      baseArrayLayer: READ_BASE_LAYER,
-      arrayLayerCount: 1,
+        // Two texture usages are binding to different mip levels of the same texture.
+        {
+          baseLevel: BASE_LEVEL + 1,
+          baseLayer: BASE_LAYER,
+          levelCount: 1,
+          layerCount: 1,
+          _resourceSuccess: true,
+        },
+
+        // Two texture usages are binding to different array layers of the same texture.
+        {
+          baseLevel: BASE_LEVEL,
+          baseLayer: BASE_LAYER + 1,
+          levelCount: 1,
+          layerCount: 1,
+          _resourceSuccess: true,
+        },
+
+        // The second texture usage contains the whole mip chain where the first texture usage is using.
+        {
+          baseLevel: 0,
+          baseLayer: BASE_LAYER,
+          levelCount: TOTAL_LEVELS,
+          layerCount: 1,
+          _resourceSuccess: false,
+        },
+
+        // The second texture usage contains the all layers where the first texture usage is using.
+        {
+          baseLevel: BASE_LEVEL,
+          baseLayer: 0,
+          levelCount: 1,
+          layerCount: TOTAL_LAYERS,
+          _resourceSuccess: false,
+        },
+      ])
+      .combine([
+        {
+          type0: 'sampled-texture',
+          type1: 'sampled-texture',
+          _usageSuccess: true,
+        },
+        {
+          type0: 'sampled-texture',
+          type1: 'readonly-storage-texture',
+          _usageSuccess: true,
+        },
+        {
+          type0: 'sampled-texture',
+          type1: 'writeonly-storage-texture',
+          _usageSuccess: false,
+        },
+        {
+          type0: 'sampled-texture',
+          type1: 'render-target',
+          _usageSuccess: false,
+        },
+        {
+          type0: 'readonly-storage-texture',
+          type1: 'readonly-storage-texture',
+          _usageSuccess: true,
+        },
+        {
```

Comment:
Based on the rules in #994 this will actually be allowed, which I think makes sense.